src/io/src/DataLoaderBridge.cpp

```diff
@@ -0,0 +1,274 @@
+#include <rs/io/DataLoaderBridge.h>
+
+//PCL include
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl/io/ply_io.h>
+
+#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem/path.hpp>
+#include <boost/progress.hpp>
+#include <boost/optional/optional.hpp>
+
+#include <functional>
+#include <iterator>
+#include <sstream>
+
+
+using namespace uima;
+
+namespace fs = boost::filesystem;
+
+DataLoaderBridge::DataLoaderBridge(const boost::property_tree::ptree& pt) : CamInterface(pt){
+  isLoop = true;
+
+  isCloudFile = true;
+  isRGBFile = true;
+  isDepthFile = true;
+
+  haveCloud = false;
+  haveRGB = false;
+  haveDepth = false;
+
+  iterator = 0;
+
+  _newData = true;
+
+  this->readConfig(pt);
+
+  if (this->frameRate > 0) {
+    auto worker = std::bind(&DataLoaderBridge::updateTimerWorker, this,
+      std::chrono::milliseconds(std::lround(1000 / this->frameRate)));
+    this->updateTimerThread = std::thread(worker);
+  }
+}
+
+DataLoaderBridge::~DataLoaderBridge() {}
+
+//NOTE: check if all are files or all size of data are equal
+bool DataLoaderBridge::checkConsistency(){
+  if(haveCloud && haveRGB){
+    if(isCloudFile && isRGBFile) return true;
+    else if(!isCloudFile && !isRGBFile){
+      if(clouds.size() == images.size())
+        return true;
+    }
+    return false;
+  }
+  else if(haveCloud && haveDepth){
+    if(isCloudFile && isDepthFile) return true;
+    else if(!isCloudFile && !isDepthFile){
+      if(clouds.size() == depths.size())
+        return true;
+    }
+    return false;
+  }
+  else if(haveRGB && haveDepth){
+    if(isRGBFile && isDepthFile) return true;
+    else if(!isRGBFile && !isDepthFile){
+      if(images.size() == depths.size())
+        return true;
+    }
+    return false;
+  }
+  else if(haveCloud && haveRGB && haveDepth){
+    if(isCloudFile && isRGBFile && isDepthFile) return true;
+    else if(!isCloudFile && !isRGBFile && !isDepthFile){
+      if(clouds.size() == images.size() && clouds.size() == depths.size())
+        return true;
+    }
+    return false;
+  }
+  return true;
```

Comment:
yeah, I think it should be for safety. But this is the entry point of the pipeline, the user has to control the validity of input data.